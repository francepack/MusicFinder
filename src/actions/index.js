export const storeBand = (band) => ({
  type: 'STORE_BAND',
  band
})

export const storeSimilarBands = (bands) => ({
  type: 'STORE_SIMILAR',
  bands
})

export const storeBandTags = (tags) => ({
  type: 'STORE_TAGS',
  tags
})

export const storeEvents = (events) => ({
  type: 'STORE_EVENTS',
  events
})