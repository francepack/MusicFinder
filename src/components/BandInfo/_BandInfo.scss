.keywords {
  display: flex;
  justify-content: space-around;
}

ul {
  list-style-type: none;
}

.list-head {
  text-decoration: underline;
  font-size: 150%;
}