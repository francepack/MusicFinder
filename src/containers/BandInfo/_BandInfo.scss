.search-area {
  color: $darkPlat;
  padding-bottom: 40px;
}

.search-area span {
  text-decoration: underline;
  &:hover {
    color: $lavender;
    cursor: pointer;
  }
}

.keywords {
  display: flex;
  justify-content: space-around;
}

ul {
  align-items: center;
  // border: 3px solid $wood;
  display: flex;
  flex-direction: column;
  list-style-type: none;
  margin: 0;
  padding: 0;
  width: 50%
}

.search-term {
  margin-bottom: 5px;
  &:hover {
    color: $lavender;
    cursor: pointer;
  }
}

.list-head {
  font-size: 150%;
  margin-bottom: 15px;
  text-decoration: underline;
}