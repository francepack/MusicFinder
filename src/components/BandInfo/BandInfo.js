import React, { Component } from 'react'
import { connect } from 'react-redux'
import { withRouter } from 'react-router-dom'
import PropTypes from 'prop-types'
import { SearchParams } from '../SearchParams/SearchParams'
import { storeEvents } from '../../actions'
import { makeUrlString, makeDateUrl, cleanEvents } from '../../helpers/infoCleaners'
import { getEvents } from '../../helpers/apiCalls'

export class BandInfo extends Component {
  constructor(props) {
    super(props)
    this.state = {
      state: '',
      city: '',
      startDate: '',
      endDate: ''
    }
  }

  collectSearchParams = (param, value) => {
    this.setState({ [param]: value })
  }

  renderKeywords = (keywords) => {
    return keywords.map((keyword, i) => {
      return(
        <li key={keyword + i}> 
          <div className='search-term' 
              onClick={() => this.handleClick(keyword)}>
            {keyword}
          </div>
        </li>
      )
    })
  }

  handleClick = (searchItem) => {
    this.searchKeyword(searchItem)
  }

  searchKeyword = async (searchItem) => {
    let url = this.createUrl(searchItem)
    try {
      let events = await this.findEvents(url)
      this.saveEvents(events)
      this.showEvents()
    } catch(error) {
      return error.message
    }
  }

  showEvents = () => {
    const { history } = this.props
    history.push('/events')
  }

  createUrl = (keyword) => {
    const { state, city, startDate, endDate } = this.state
    const keywordUrl = makeUrlString(keyword)
    let urlString = `classificationName=music&keyword=${keywordUrl}`
    if (state) {
      const cleanState = makeUrlString(state)
      const stateUrl = `&stateCode=${cleanState}`
      urlString = urlString + stateUrl
    }
    if (city) {
      const cleanCity = makeUrlString(city)
      const cityUrl = `&city=${cleanCity}`
      urlString = urlString + cityUrl
    }
    return urlString
  }

  findEvents = async (url) => {
    try {
      let events = await getEvents(url)
      return events
    } catch (error) {
      return error
    }
  }

  saveEvents = (events) => {
    this.props.storeEvents(events)
  }

  render() {
    const similarBands = this.renderKeywords(this.props.similarBands)
    const tags = this.renderKeywords(this.props.tags)
    return(
      <div className='search-area'>
        <SearchParams collectSearchParams={this.collectSearchParams} />
        <h4>Click a search term below</h4>
        <p>You searched: <span onClick={() => this.handleClick(this.props.band)}>{this.props.band}</span></p>
        <div className='keywords'>
          <ul className='bands-list'>
            <li className='list-head'>Similar Bands</li>
            {similarBands}
          </ul>
          <ul className='tags-list'>
            <li className='list-head'>Associated Terms</li>
            {tags}
          </ul>
        </div>
      </div>
    )
  }
}

export const mapStateToProps = (state) => ({
  band: state.band
})

export const mapDispatchToProps = (dispatch) => ({
  storeEvents: (events) => dispatch(storeEvents(events))
})

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(BandInfo))