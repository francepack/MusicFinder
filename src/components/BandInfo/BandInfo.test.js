import React from 'react'
import { BandInfo, mapDispatchToProps, mapStateToProps } from './BandInfo'
import { storeEvents } from '../../actions'
import { shallow } from 'enzyme'

describe('BandInfo', () => {
  let wrapper
  let props
  beforeEach(() => {
    props = {
      band: 'ween',
      similarBands: ['beatles', 'RHCP', 'Yes'],
      tags: ['weird', 'happy sometimes', 'sad sometimes'],
      storeEvents: jest.fn()
    }
    wrapper = shallow(
      <BandInfo {...props} />
    )
  })
  describe('standards', () => {
    it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    })
    it('should have default state', () => {
      const expected = {
        state: '',
        city: '',
        startDate: '',
        endDate: ''
      }
      expect(wrapper.state()).toEqual(expected)
    })
  })
  describe('', () => {
    it('should', () => {
      
    })
  })
  describe('', () => {
    it('should', () => {
      
    })
  })
  describe('', () => {
    it('should', () => {
      
    })
  })
  describe('mapDispatchToProps', () => {
    it('should store events', () => {
      const mockEvents = [{name: 'Big Gig'}, {name: 'cochella'}]
      const mockDispatch = jest.fn()
      const actionToDispatch = storeEvents(mockEvents)
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.storeEvents(mockEvents)
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })
  })
  describe('mapStateToProps', () => {
    it('should return an object with a band, similar bands, and tags', () => {
      const mockState = {
        band: '',
        similarBands: [],
        tags: [],
        events: []
      }
      const expected = {
        band: '',
        similarBands: [],
        tags: []
      }
      const result = mapStateToProps(mockState)
      expect(result).toEqual(expected)
    })
  })
})