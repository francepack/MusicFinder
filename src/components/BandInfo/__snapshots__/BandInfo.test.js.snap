// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BandInfo standards should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BandInfo
    band="ween"
    setLoading={[MockFunction]}
    similarBands={
      Array [
        "beatles",
        "RHCP",
        "Yes",
      ]
    }
    storeEvents={[MockFunction]}
    tags={
      Array [
        "weird",
        "happy sometimes",
        "sad sometimes",
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <SearchParams
          collectSearchParams={[Function]}
        />,
        <h4>
          Click a search term below
        </h4>,
        <p>
          You searched: 
          <span
            onClick={[Function]}
          >
            ween
          </span>
        </p>,
        <div
          className="keywords"
        >
          <ul
            className="bands-list"
          >
            <li
              className="list-head"
            >
              Similar Bands
            </li>
            <li>
              <div
                className="search-term"
                onClick={[Function]}
              >
                beatles
              </div>
            </li>
            <li>
              <div
                className="search-term"
                onClick={[Function]}
              >
                RHCP
              </div>
            </li>
            <li>
              <div
                className="search-term"
                onClick={[Function]}
              >
                Yes
              </div>
            </li>
          </ul>
          <ul
            className="tags-list"
          >
            <li
              className="list-head"
            >
              Associated Terms
            </li>
            <li>
              <div
                className="search-term"
                onClick={[Function]}
              >
                weird
              </div>
            </li>
            <li>
              <div
                className="search-term"
                onClick={[Function]}
              >
                happy sometimes
              </div>
            </li>
            <li>
              <div
                className="search-term"
                onClick={[Function]}
              >
                sad sometimes
              </div>
            </li>
          </ul>
        </div>,
      ],
      "className": "search-area",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "collectSearchParams": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Click a search term below",
        },
        "ref": null,
        "rendered": "Click a search term below",
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "You searched: ",
            <span
              onClick={[Function]}
            >
              ween
            </span>,
          ],
        },
        "ref": null,
        "rendered": Array [
          "You searched: ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "ween",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "ween",
            "type": "span",
          },
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <ul
              className="bands-list"
            >
              <li
                className="list-head"
              >
                Similar Bands
              </li>
              <li>
                <div
                  className="search-term"
                  onClick={[Function]}
                >
                  beatles
                </div>
              </li>
              <li>
                <div
                  className="search-term"
                  onClick={[Function]}
                >
                  RHCP
                </div>
              </li>
              <li>
                <div
                  className="search-term"
                  onClick={[Function]}
                >
                  Yes
                </div>
              </li>
            </ul>,
            <ul
              className="tags-list"
            >
              <li
                className="list-head"
              >
                Associated Terms
              </li>
              <li>
                <div
                  className="search-term"
                  onClick={[Function]}
                >
                  weird
                </div>
              </li>
              <li>
                <div
                  className="search-term"
                  onClick={[Function]}
                >
                  happy sometimes
                </div>
              </li>
              <li>
                <div
                  className="search-term"
                  onClick={[Function]}
                >
                  sad sometimes
                </div>
              </li>
            </ul>,
          ],
          "className": "keywords",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <li
                  className="list-head"
                >
                  Similar Bands
                </li>,
                Array [
                  <li>
                    <div
                      className="search-term"
                      onClick={[Function]}
                    >
                      beatles
                    </div>
                  </li>,
                  <li>
                    <div
                      className="search-term"
                      onClick={[Function]}
                    >
                      RHCP
                    </div>
                  </li>,
                  <li>
                    <div
                      className="search-term"
                      onClick={[Function]}
                    >
                      Yes
                    </div>
                  </li>,
                ],
              ],
              "className": "bands-list",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Similar Bands",
                  "className": "list-head",
                },
                "ref": null,
                "rendered": "Similar Bands",
                "type": "li",
              },
              Object {
                "instance": null,
                "key": "beatles0",
                "nodeType": "host",
                "props": Object {
                  "children": <div
                    className="search-term"
                    onClick={[Function]}
                  >
                    beatles
                  </div>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "beatles",
                    "className": "search-term",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "beatles",
                  "type": "div",
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": "RHCP1",
                "nodeType": "host",
                "props": Object {
                  "children": <div
                    className="search-term"
                    onClick={[Function]}
                  >
                    RHCP
                  </div>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "RHCP",
                    "className": "search-term",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "RHCP",
                  "type": "div",
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": "Yes2",
                "nodeType": "host",
                "props": Object {
                  "children": <div
                    className="search-term"
                    onClick={[Function]}
                  >
                    Yes
                  </div>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Yes",
                    "className": "search-term",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Yes",
                  "type": "div",
                },
                "type": "li",
              },
            ],
            "type": "ul",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <li
                  className="list-head"
                >
                  Associated Terms
                </li>,
                Array [
                  <li>
                    <div
                      className="search-term"
                      onClick={[Function]}
                    >
                      weird
                    </div>
                  </li>,
                  <li>
                    <div
                      className="search-term"
                      onClick={[Function]}
                    >
                      happy sometimes
                    </div>
                  </li>,
                  <li>
                    <div
                      className="search-term"
                      onClick={[Function]}
                    >
                      sad sometimes
                    </div>
                  </li>,
                ],
              ],
              "className": "tags-list",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Associated Terms",
                  "className": "list-head",
                },
                "ref": null,
                "rendered": "Associated Terms",
                "type": "li",
              },
              Object {
                "instance": null,
                "key": "weird0",
                "nodeType": "host",
                "props": Object {
                  "children": <div
                    className="search-term"
                    onClick={[Function]}
                  >
                    weird
                  </div>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "weird",
                    "className": "search-term",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "weird",
                  "type": "div",
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": "happy sometimes1",
                "nodeType": "host",
                "props": Object {
                  "children": <div
                    className="search-term"
                    onClick={[Function]}
                  >
                    happy sometimes
                  </div>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "happy sometimes",
                    "className": "search-term",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "happy sometimes",
                  "type": "div",
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": "sad sometimes2",
                "nodeType": "host",
                "props": Object {
                  "children": <div
                    className="search-term"
                    onClick={[Function]}
                  >
                    sad sometimes
                  </div>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "sad sometimes",
                    "className": "search-term",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "sad sometimes",
                  "type": "div",
                },
                "type": "li",
              },
            ],
            "type": "ul",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <SearchParams
            collectSearchParams={[Function]}
          />,
          <h4>
            Click a search term below
          </h4>,
          <p>
            You searched: 
            <span
              onClick={[Function]}
            >
              ween
            </span>
          </p>,
          <div
            className="keywords"
          >
            <ul
              className="bands-list"
            >
              <li
                className="list-head"
              >
                Similar Bands
              </li>
              <li>
                <div
                  className="search-term"
                  onClick={[Function]}
                >
                  beatles
                </div>
              </li>
              <li>
                <div
                  className="search-term"
                  onClick={[Function]}
                >
                  RHCP
                </div>
              </li>
              <li>
                <div
                  className="search-term"
                  onClick={[Function]}
                >
                  Yes
                </div>
              </li>
            </ul>
            <ul
              className="tags-list"
            >
              <li
                className="list-head"
              >
                Associated Terms
              </li>
              <li>
                <div
                  className="search-term"
                  onClick={[Function]}
                >
                  weird
                </div>
              </li>
              <li>
                <div
                  className="search-term"
                  onClick={[Function]}
                >
                  happy sometimes
                </div>
              </li>
              <li>
                <div
                  className="search-term"
                  onClick={[Function]}
                >
                  sad sometimes
                </div>
              </li>
            </ul>
          </div>,
        ],
        "className": "search-area",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "collectSearchParams": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Click a search term below",
          },
          "ref": null,
          "rendered": "Click a search term below",
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "You searched: ",
              <span
                onClick={[Function]}
              >
                ween
              </span>,
            ],
          },
          "ref": null,
          "rendered": Array [
            "You searched: ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "ween",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "ween",
              "type": "span",
            },
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <ul
                className="bands-list"
              >
                <li
                  className="list-head"
                >
                  Similar Bands
                </li>
                <li>
                  <div
                    className="search-term"
                    onClick={[Function]}
                  >
                    beatles
                  </div>
                </li>
                <li>
                  <div
                    className="search-term"
                    onClick={[Function]}
                  >
                    RHCP
                  </div>
                </li>
                <li>
                  <div
                    className="search-term"
                    onClick={[Function]}
                  >
                    Yes
                  </div>
                </li>
              </ul>,
              <ul
                className="tags-list"
              >
                <li
                  className="list-head"
                >
                  Associated Terms
                </li>
                <li>
                  <div
                    className="search-term"
                    onClick={[Function]}
                  >
                    weird
                  </div>
                </li>
                <li>
                  <div
                    className="search-term"
                    onClick={[Function]}
                  >
                    happy sometimes
                  </div>
                </li>
                <li>
                  <div
                    className="search-term"
                    onClick={[Function]}
                  >
                    sad sometimes
                  </div>
                </li>
              </ul>,
            ],
            "className": "keywords",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <li
                    className="list-head"
                  >
                    Similar Bands
                  </li>,
                  Array [
                    <li>
                      <div
                        className="search-term"
                        onClick={[Function]}
                      >
                        beatles
                      </div>
                    </li>,
                    <li>
                      <div
                        className="search-term"
                        onClick={[Function]}
                      >
                        RHCP
                      </div>
                    </li>,
                    <li>
                      <div
                        className="search-term"
                        onClick={[Function]}
                      >
                        Yes
                      </div>
                    </li>,
                  ],
                ],
                "className": "bands-list",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Similar Bands",
                    "className": "list-head",
                  },
                  "ref": null,
                  "rendered": "Similar Bands",
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": "beatles0",
                  "nodeType": "host",
                  "props": Object {
                    "children": <div
                      className="search-term"
                      onClick={[Function]}
                    >
                      beatles
                    </div>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "beatles",
                      "className": "search-term",
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": "beatles",
                    "type": "div",
                  },
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": "RHCP1",
                  "nodeType": "host",
                  "props": Object {
                    "children": <div
                      className="search-term"
                      onClick={[Function]}
                    >
                      RHCP
                    </div>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "RHCP",
                      "className": "search-term",
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": "RHCP",
                    "type": "div",
                  },
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": "Yes2",
                  "nodeType": "host",
                  "props": Object {
                    "children": <div
                      className="search-term"
                      onClick={[Function]}
                    >
                      Yes
                    </div>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Yes",
                      "className": "search-term",
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": "Yes",
                    "type": "div",
                  },
                  "type": "li",
                },
              ],
              "type": "ul",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <li
                    className="list-head"
                  >
                    Associated Terms
                  </li>,
                  Array [
                    <li>
                      <div
                        className="search-term"
                        onClick={[Function]}
                      >
                        weird
                      </div>
                    </li>,
                    <li>
                      <div
                        className="search-term"
                        onClick={[Function]}
                      >
                        happy sometimes
                      </div>
                    </li>,
                    <li>
                      <div
                        className="search-term"
                        onClick={[Function]}
                      >
                        sad sometimes
                      </div>
                    </li>,
                  ],
                ],
                "className": "tags-list",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Associated Terms",
                    "className": "list-head",
                  },
                  "ref": null,
                  "rendered": "Associated Terms",
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": "weird0",
                  "nodeType": "host",
                  "props": Object {
                    "children": <div
                      className="search-term"
                      onClick={[Function]}
                    >
                      weird
                    </div>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "weird",
                      "className": "search-term",
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": "weird",
                    "type": "div",
                  },
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": "happy sometimes1",
                  "nodeType": "host",
                  "props": Object {
                    "children": <div
                      className="search-term"
                      onClick={[Function]}
                    >
                      happy sometimes
                    </div>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "happy sometimes",
                      "className": "search-term",
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": "happy sometimes",
                    "type": "div",
                  },
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": "sad sometimes2",
                  "nodeType": "host",
                  "props": Object {
                    "children": <div
                      className="search-term"
                      onClick={[Function]}
                    >
                      sad sometimes
                    </div>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "sad sometimes",
                      "className": "search-term",
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": "sad sometimes",
                    "type": "div",
                  },
                  "type": "li",
                },
              ],
              "type": "ul",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
